### DISCUSSION TITLE
How can I add Adorner to my usercontrol?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#10752] [Q&A] [answered]
I am migrating an existing WPF personal project to Avalonia.

Previously in my project I added an Adorner to a usercontrol.

However, it seems Avalonia has changed Adorner to AdornerLayer. I am not quite sure about this. What's more, no matter in Google or Avalonia official site, there is not any documentation or tutorial about this.

Would you please give me a sample? Thank you.

**Comment by wieslawsoltes** [comment, marked as answer]
In Avalonia 11.0.0 latest preview

https://github.com/AvaloniaUI/Avalonia/blob/74d0241e7fefe38d1bfcdbac1a95c2dd50a30578/samples/ControlCatalog/Pages/AdornerLayerPage.xaml#L46

https://github.com/AvaloniaUI/Avalonia/blob/74d0241e7fefe38d1bfcdbac1a95c2dd50a30578/samples/ControlCatalog/Pages/AdornerLayerPage.xaml.cs#L32

**Reply by wieslawsoltes** [reply]
https://github.com/AvaloniaUI/Avalonia/blob/74d0241e7fefe38d1bfcdbac1a95c2dd50a30578/src/Avalonia.Controls/Primitives/AdornerLayer.cs#L169-L192

**Reply by kyuranger** [reply]
> In Avalonia 11.0.0 latest preview
> 
> https://github.com/AvaloniaUI/Avalonia/blob/74d0241e7fefe38d1bfcdbac1a95c2dd50a30578/samples/ControlCatalog/Pages/AdornerLayerPage.xaml#L46
> 
> https://github.com/AvaloniaUI/Avalonia/blob/74d0241e7fefe38d1bfcdbac1a95c2dd50a30578/samples/ControlCatalog/Pages/AdornerLayerPage.xaml.cs#L32

Thanks a lot. Let's me try this.
